internal/testutil/assert.go

```diff
@@ -0,0 +1,165 @@
+package testutil
+
+import (
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/newrelic/infra-integrations-sdk/data/metric"
+	"github.com/newrelic/infra-integrations-sdk/integration"
+	"github.com/newrelic/nri-kubernetes/v2/src/definition"
+)
+
+// Asserter is a helper for checking whether an integration contains all the metrics defined in a specGroup.
+// It provides a chainable API.
+// Asserter is safe to use concurrently.
+type Asserter struct {
+	*sync.Mutex
+	entities        []*integration.Entity
+	specGroups      definition.SpecGroups
+	exclude         map[string]map[string]bool
+	excludeOptional bool
+	silent          bool
+}
+
+func NewAsserter() Asserter {
+	return Asserter{
+		Mutex: &sync.Mutex{},
+	}
+}
+
+// Using returns an asserter that will use the supplied specGroups to assert entities.
+func (a Asserter) Using(groups definition.SpecGroups) Asserter {
+	a.specGroups = groups
```

Comment:
does it make sense to run any of the group without calling "Using"? Should we have it as a parameter of the builder?

Reply:
Or of the asserter

Reply:
Doing it in the builder is an option, but I wanted to keep everything through chainable methods. In fact in an previous iteration it _was_ in the `.Assert` method, but then you had to pass it several times. Like this, you can configure the asserter first with the groups (that will likely be constant across the whole test) and then just add the variable things (entities) after the test fans out in the tables and so on. This is also the reason to why the `Asserter` returns a copy of itself with each chainable method.

Reply:
I have clarified the comments on how the asserter work, as well as making it more explicit in this test since it's pretty simple (for now).